client/src/entrypoints/snippets/snippet-chooser.js

```diff
@@ -1,95 +1,26 @@
-  createSnippetChooser. State is either null (= no document chosen) or a dict of id, string and
-  edit_link.
-
-  The result returned from the snippet chooser modal (see wagtail.snippets.views.chooser.chosen)
-  is a superset of this, and can therefore be passed directly to chooser.setState.
-  */
-  let state = null;
-  if (input.val()) {
-    state = {
-      id: input.val(),
-      edit_link: editLink.attr('href'),
-      string: docTitle.text(),
-    };
+class SnippetChooser extends Chooser {
+  titleStateKey = 'string';
+
+  getModalUrl() {
+    let urlQuery = '';
+    if (wagtailConfig.ACTIVE_CONTENT_LOCALE) {
+      // The user is editing a piece of translated content.
+      // Pass the locale along as a request parameter. If this
+      // snippet is also translatable, the results will be
+      // pre-filtered by this locale.
+      urlQuery = '?locale=' + wagtailConfig.ACTIVE_CONTENT_LOCALE;
+    }
+    return this.chooserBaseUrl + urlQuery;
   }
+}
+window.SnippetChooser = SnippetChooser;
 
-  /* define public API functions for the chooser */
-  const chooser = {
-    getState: () => state,
-    getValue: () => state && state.id,
-    setState: (newState) => {
-      if (newState) {
-        input.val(newState.id);
-        docTitle.text(newState.string);
-        chooserElement.removeClass('blank');
-        editLink.attr('href', newState.edit_link);
-      } else {
-        input.val('');
-        chooserElement.addClass('blank');
-      }
-
-      state = newState;
-    },
-    getTextLabel: (opts) => {
-      if (!state) return null;
-      const result = state.string;
-      if (opts && opts.maxLength && result.length > opts.maxLength) {
-        return result.substring(0, opts.maxLength - 1) + '…';
-      }
-      return result;
-    },
-    focus: () => {
-      $('.action-choose', chooserElement).focus();
-    },
-    openChooserModal: () => {
-      let urlQuery = '';
-      if (wagtailConfig.ACTIVE_CONTENT_LOCALE) {
-        // The user is editing a piece of translated content.
-        // Pass the locale along as a request parameter. If this
-        // snippet is also translatable, the results will be
-        // pre-filtered by this locale.
-        urlQuery = '?locale=' + wagtailConfig.ACTIVE_CONTENT_LOCALE;
-      }
-
-      // eslint-disable-next-line no-undef
-      ModalWorkflow({
-        url: chooserBaseUrl + urlQuery,
-        // eslint-disable-next-line no-undef
-        onload: SNIPPET_CHOOSER_MODAL_ONLOAD_HANDLERS,
-        responses: {
-          snippetChosen: (result) => {
-            chooser.setState(result);
-          },
-        },
-      });
-    },
-
-    clear: () => {
-      chooser.setState(null);
-    },
-  };
-
-  $('.action-choose', chooserElement).on('click', () => {
-    chooser.openChooserModal();
-  });
-
-  $('.action-clear', chooserElement).on('click', () => {
-    chooser.clear();
-  });
-
-  return chooser;
+function createSnippetChooser(id) {
+  /* RemovedInWagtail50Warning */
```

Comment:
Will this comment actually do anything or is it just a comment for future reference?

Should we look at adding some clientside error classes for things like this?

Reply:
Yep, it's just a comment for me (or whoever does the 5.0 release management) to find when we're removing deprecated features and searching for that string.

I'd be happy to hear suggestions for how to better handle deprecations in client-side code... I guess we can't really make them as visible as server-side ones, as I suspect people don't routinely look in the browser console when upgrading, but it can't hurt.

Reply:
yeah - had a think about this and we would need to wrangle things a bit to ensure the error did not throw (break JS code) and it also assumes they are looking at the console.

Probably simpler to leave as is for now.

Reply:
Created an issue - I have been thinking about this and think it would be valuable. https://github.com/wagtail/wagtail/issues/8956